### PR TITLE
feat: add repost button that handles post to blog

### PR DESCRIPTION
**Pull request by jackson-harris-iii**
All of the reposts work is in progress, this commit is to close out the blog post ui changes [issue](https://github.com/near/near-discovery/issues/993), however the work is inside of the "Reposts" button and the repost work itself is incomplete.

Closes: https://github.com/near/near-discovery/issues/993

### CODE REVIEW

File: src/Posts/RepostButton.jsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  (reposts || []).filter((repost) => repost.value.type === "repost").map((repost) => [repost.accountId, repost]),
+);
+
+if (state.hasRepost === true) {
+  repostsByUsers[context.accountId] = {
+    accountId: context.accountId,
+  };
+}
+
+const accountsWithReposts = Object.keys(repostsByUsers);
```
Comment (charleslavon) [inline src/Posts/RepostButton.jsx:21]: `accountsWithReposts ` is unused

File: src/Posts/RepostButton.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+const hasRepost = context.accountId && !!repostsByUsers[context.accountId];
```
Comment (charleslavon) [inline src/Posts/RepostButton.jsx:22]: Curious to know the intention behind a `state.hasRepost` and this locally scoped `hasRepost`

File: src/Posts/RepostButton.jsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+const reposts = Social.index("repost", item);
+
+const dataLoading = reposts === null;
+
+const repostsByUsers = Object.fromEntries(
+  (reposts || []).filter((repost) => repost.value.type === "repost").map((repost) => [repost.accountId, repost]),
+);
+
+if (state.hasRepost === true) {
+  repostsByUsers[context.accountId] = {
```
Comment (charleslavon) [inline src/Posts/RepostButton.jsx:16]: This AccountId-to-repost object is being converted to a AccountId-to-String. Am I understanding this correctly? 
However we're only using this object to check if a truthy value exists. If that's true, is this assignment necessary?

File: src/Posts/RepostButton.jsx (hunk 1 above)
Comment (jackson-harris-iii) [reply src/Posts/RepostButton.jsx]: All of the reposts work is in progress, this commit is to close out the blog post ui changes (issue)[https://github.com/near/near-discovery/issues/993], however the work is inside of the "Reposts" button and the repost work itself is incomplete.

File: src/Posts/RepostButton.jsx (hunk 2 above)
Comment (jackson-harris-iii) [reply src/Posts/RepostButton.jsx]: see updated commit notes.

File: src/Posts/RepostButton.jsx (hunk 3 above)
Comment (jackson-harris-iii) [reply src/Posts/RepostButton.jsx]: see updated commit notes about reposts as WIP

**Review by charleslavon — APPROVED** [review]

**Pull request merged**